lexicon/providers/dinahosting.py

```diff
@@ -0,0 +1,258 @@
+RC_OBJECT_NOT_EXISTS = 2303
+RC_COMMAND_FAILED = 2400
+RC_COMMAND_FAILED_FATAL = 2500
+RC_COMMAND_TIMEOUT = 2501
+
+
+def provider_parser(subparser):
+    """Return the parser for this provider"""
+    subparser.add_argument(
+        "--auth-username", help="specify username for authentication")
+    subparser.add_argument(
+        "--auth-password", help="specify password for authentication")
+
+
+class Provider(BaseProvider):
+    """Provider class for Dinahosting"""
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.domain_id = None
+        self.api_endpoint = 'https://dinahosting.com/special/api.php'
+
+    def _authenticate(self):
+        data = {
+            'command': 'Services_GetDomains'
+        }
+        payload = self._post('', data)
+
+        if not payload['data']:
+            raise Exception('No domain found')
+        domain_found = False
+        for data in payload['data']:
+            if data['domain'] == self.domain:
+                domain_found = True
+                break
+        if not domain_found:
+            raise Exception('Requested domain is not among the owned domains')
+
+        self.domain_id = self.domain
+
+    def _create_record(self, rtype, name, content):
+        self._check_unsupported_type(rtype)
+
+        rel_name = self._relative_name(name)
+        data = {
+            'domain': self.domain,
+            'hostname': rel_name
+        }
+
+        if rtype == 'A':
+            data['command'] = "Domain_Zone_AddTypeA"
+            data['ip'] = content
+        elif rtype == 'AAAA':
+            data['command'] = "Domain_Zone_AddTypeAAAA"
+            data['ip'] = content
+        elif rtype == 'CNAME':
+            data['command'] = "Domain_Zone_AddTypeCname"
+            data['destinationHostname'] = content
+        elif rtype == 'TXT':
+            data['command'] = "Domain_Zone_AddTypeTXT"
+            data['text'] = content
+
+        try:
+            self._post('', data)
+        except requests.exceptions.HTTPError as err:
+            already_exists = ((err.response.json()['responseCode'] == RC_OBJECT_EXISTS) or
+                              (err.response.json()['responseCode'] == RC_COMMAND_FAILED))
+            if not already_exists:
+                raise
+
+        LOGGER.debug('create_record: %s', True)
+        return True
+
+    def _list_records(self, rtype=None, name=None, content=None):
+        if rtype:
+            formatted_rtype = "Cname" if rtype == 'CNAME' else rtype
+            data = {
+                'command': 'Domain_Zone_GetType{0}'.format(formatted_rtype),
+                'domain': self.domain
+            }
+        else:
+            data = {
+                'command': 'Domain_Zone_GetAll',
+                'domain': self.domain
+            }
+
+        payload = self._post('', data)
+
+        #content_obj = self._parse_record_content(rtype, content)
+        #content = content_obj['data']
```

Comment:
Some commented code to remove.

Reply:
Agreed. While I was at it I modified the _list_records to correct the lint issue and followed one of your suggestions from another PR about extracting the pseudo-id to its own _identifier method.